### Ejercicios con archivoss

#### Trabajando con archivo dna.txt en ruta `"../data/dna.txt"`

In [6]:
# Abriendo el archivo
#my_file = open("../data/dna.txt")
# Leyendo el archivo
#my_file_contents = my_file.read()

# Agregar el nombre del archivo como variable
my_file_name = "../data/dna.txt"

# Leer el archivo
my_file = open(my_file_name)
my_dna = my_file.read()

# Calular la longitud
dna_length = len(my_dna)

# Mostrar mensaje
print("The sequence is " + my_dna +  " and length is " + str(dna_length))

The sequence is ACCCGCGTATGCACACTGCATGGGGGTCACTGCTAATCAAACGCCGTAGGTTCAATACCGATTGCGCCCAAATCACCTGCCGAAATCGACGATTAGTAATGCCTATTCGAATTACAATATGCCGCGTCGGTTGATCCGAATAGTGTATCATAAAACTACACGTTTGAGATAGAAAGAGGGCTTAGGCGCGTACGCAATAAGCGATAGTGGTACAGGTGCTCTTAACCAAGTCTGTCAGGGCCTACCCTTCTCTGGTATGGTTTGCAAGGTCGGTCCTACATTCCTACTTTATGCCAGTGGTGCAGTTTATTCAGGTTGCAGGCATATAGGGTATTGTGCTTGGCAGGTCGCGTAAAAAGCGCTCGCATTTTACCATTATGATGCCTATGACTTTCCAGTGGAGGCATCTAAAAGTGGGCACGAATTGTTTTTTGGTTGTCGTTAAACACCAATGTGTATAGGGCCTGGAGATAGAGGGGGCCTATACAGTTTCGAAGTGGTGACATCCAACGCCCAGGCCTCCCGCTTGGTGGAAAAGCGTACGGAGCTCTAAATACTCCAAGATCTAAACCACTTGGCTGAGGGTATGCTATCTGTCCTATCACCAACGCAACCGTGCATGGAAGGATAGGCGCGAGCGCCTAGTAAGCTCGAATCAGTGTAATTTGCGAAGCGCTCGCTATGCGTATGCGACTGTGTGAGTTGCAGGTAGAAGACAAATTCAATCGTCGTGGGCGGTTACTCCTCTACAAATAACGTCCTGCAGAATTACGACCCCGCCGAAACTCTAGCAGTTGGGGGTTGTCTAACCGAGGGTACCGGAATTCACCGAAATGTT
 and length is 839


Cuenta 839 caracteres pero python lee el archivo con un \n por lo tanto este salto de línea afecta el conteo, se quita con la siguiente línea

In [7]:
# Quitar el caracter de nueva linea \n
my_dna = my_file_contents.rstrip("\n")

# O podemos my_dna = my_file.read().rstrip("\n")

# Calular la longitud
dna_length = len(my_dna)

# Mostrar mensaje
print("The sequence is " + my_dna +  " and length is " + str(dna_length))

The sequence is ACCCGCGTATGCACACTGCATGGGGGTCACTGCTAATCAAACGCCGTAGGTTCAATACCGATTGCGCCCAAATCACCTGCCGAAATCGACGATTAGTAATGCCTATTCGAATTACAATATGCCGCGTCGGTTGATCCGAATAGTGTATCATAAAACTACACGTTTGAGATAGAAAGAGGGCTTAGGCGCGTACGCAATAAGCGATAGTGGTACAGGTGCTCTTAACCAAGTCTGTCAGGGCCTACCCTTCTCTGGTATGGTTTGCAAGGTCGGTCCTACATTCCTACTTTATGCCAGTGGTGCAGTTTATTCAGGTTGCAGGCATATAGGGTATTGTGCTTGGCAGGTCGCGTAAAAAGCGCTCGCATTTTACCATTATGATGCCTATGACTTTCCAGTGGAGGCATCTAAAAGTGGGCACGAATTGTTTTTTGGTTGTCGTTAAACACCAATGTGTATAGGGCCTGGAGATAGAGGGGGCCTATACAGTTTCGAAGTGGTGACATCCAACGCCCAGGCCTCCCGCTTGGTGGAAAAGCGTACGGAGCTCTAAATACTCCAAGATCTAAACCACTTGGCTGAGGGTATGCTATCTGTCCTATCACCAACGCAACCGTGCATGGAAGGATAGGCGCGAGCGCCTAGTAAGCTCGAATCAGTGTAATTTGCGAAGCGCTCGCTATGCGTATGCGACTGTGTGAGTTGCAGGTAGAAGACAAATTCAATCGTCGTGGGCGGTTACTCCTCTACAAATAACGTCCTGCAGAATTACGACCCCGCCGAAACTCTAGCAGTTGGGGGTTGTCTAACCGAGGGTACCGGAATTCACCGAAATGTT and length is 838


#### Ejercicio de creación de archivos

In [24]:
# Creación de archivo
myfile = open("../results/out.txt","w")
myfile.write("Hola este es un ejercicio para hacer un archivo")
myfile.close()

In [25]:
# Imprimir archivo creado
out=open("../results/out.txt")
print(out.read())

Hola este es un ejercicio para hacer un archivo
